.github/workflows/gh_pages.yml

```diff
@@ -1,30 +1,30 @@
 name: Github Pages
 on:
-  release:
-    types: [published]
+  push:
+    branches:
+      - master
 jobs:
   build-and-deploy:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout
-        uses: actions/checkout@v2.3.1
-      
+      - uses: actions/checkout@v2.3.1
+
       - name: Install Node (14.x)
         uses: actions/setup-node@v2
         with:
           node-version: '14.x'
 
       - name: Install Dependencies
         run: yarn install
-        
+
       - name: Build
         run: gulp
 
       - name: Generate JSDoc
-        run: yarn jsdoc -c .jsdoc.json --verbose
+        run: yarn docs
 
       - name: Deploy to GH Pages
-        uses: JamesIves/github-pages-deploy-action@0f24da7de3e7e135102609a4c9633b025be8411b
+        uses: JamesIves/github-pages-deploy-action@4.1.5
         with:
           branch: gh-pages # The branch the action should deploy to.
           folder: jsdoc # The folder the action should deploy.
```

Comment:
I think what we're missing here is a Github Personal Access Token (PAT), according with the action documentation: https://github.com/JamesIves/github-pages-deploy-action#required-setup

Reply:
Update: I was wrong about the GH token being needed here, I've updated the gh-pages from my local machine so the permissions should be right. The problem could be in the action we're using.

TBH we shouldn't need an Action here... we could probably just use our own script to do something like this:
https://github.com/stellar/js-stellar-base/blob/a37e12558c04d26dc356ede476ac0ecb4036843f/after_deploy.sh

Reply:
Tradeoffs, right? With an action, we "outsource" the process and don't need to worry about publishing, but OTOH we have less control and it's harder to debug (like here)...

For `after_deploy`, how do we set up the `stellar-jenkins@` credential properly? I'll follow-up with ops.

Reply:
We shouldn't need that for the GH actions, it's legacy from the Travis CI. We could just use run the commands directly from the yml file, something in the lines of https://github.com/actions/checkout#push-a-commit-using-the-built-in-token:

```yml
on: push
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Prepare & Publish Github Pages
        uses: actions/checkout@v2
        run: |
          git config user.name github-actions
          git config user.email github-actions@github.com
          git clone -b gh-pages "git@github.com:stellar/js-stellar-base.git" jsdoc
          yarn docs
          git commit -m `git describe --abbrev=0 --tags`
          git push origin
```

Reply:
Oh wow, very nice. Have a look at the change now? Done in 7d42fa6, then broken down into discrete steps in 0c75b47.